tests/unit/test_snap.py

```diff
@@ -534,6 +534,12 @@ def test_hold_refresh_invalid_non_int(self):
         except ValueError as e:
             self.assertEqual(str(e), "days must be an int between 1 and 90")
 
+    def test_hold_refresh_invalid_non_bool(self):
+        try:
+            snap.hold_refresh(forever="foobar")
+        except ValueError as e:
+            self.assertEqual(str(e), "forever must be a boolean")
```

Comment:
When writing tests like this you have to assert that it did raise an error, rather than just "if it raises an error, check something". You do that like so (also note the `TypeError`):

```python
with self.assertRaises(TypeError):
    snap.hold_refresh(forever="foobar")
```

I also suggest we *don't* check the string message, as that may change and isn't part of the contract we should be testing.